Tests/NIOHTTP2Tests/ConfiguringPipelineTests.swift

```diff
@@ -181,4 +183,114 @@ class ConfiguringPipelineTests: XCTestCase {
         }
         XCTAssertNoThrow(try self.serverChannel.finish())
     }
+
+    /// A simple channel handler that records inbound frames.
+    class HTTP1ServerRequestRecorderHandler: ChannelInboundHandler {
+        typealias InboundIn = HTTPServerRequestPart
+
+        var receivedParts: [HTTPServerRequestPart] = []
+
+        func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+            self.receivedParts.append(self.unwrapInboundIn(data))
+        }
+    }
+
+    func testNegotiatedHTTP2BasicPipelineCommunicates() throws {
+        let serverRecorder = HTTP1ServerRequestRecorderHandler()
+        let clientHandler = try assertNoThrowWithValue(self.clientChannel.configureHTTP2Pipeline(mode: .client).wait())
+        XCTAssertNoThrow(try self.serverChannel.configureCommonHTTPServerPipeline { channel in
+            return channel.pipeline.addHandler(serverRecorder)
+        }.wait())
+
+        // Let's pretent the TLS handler did protocol negotiation for us
+        self.serverChannel.pipeline.fireUserInboundEventTriggered (TLSUserEvent.handshakeCompleted(negotiatedProtocol: "h2"))
+
+        XCTAssertNoThrow(try self.assertDoHandshake(client: self.clientChannel, server: self.serverChannel))
+
+        // Let's try sending an h2 request.
+        let requestPromise = self.clientChannel.eventLoop.makePromise(of: Void.self)
+        let reqFrame = HTTP2Frame(streamID: 1, payload: .headers(.init(headers: HPACKHeaders([(":method", "GET"), (":authority", "localhost"), (":scheme", "https"), (":path", "/testH2toHTTP1")]), endStream: true)))
+
+        clientHandler.createStreamChannel(promise: nil) { channel, streamID in
+            XCTAssertEqual(streamID, HTTP2StreamID(1))
+            channel.writeAndFlush(reqFrame).whenComplete { _ in channel.close(promise: requestPromise) }
+            return channel.eventLoop.makeSucceededFuture(())
+        }
+
+        // In addition to interacting in memory, we need two loop spins. The first is to execute `createStreamChannel`.
+        // The second is to execute the close promise callback, after the interaction is complete.
+        (self.clientChannel.eventLoop as! EmbeddedEventLoop).run()
+        self.interactInMemory(self.clientChannel, self.serverChannel)
+        (self.clientChannel.eventLoop as! EmbeddedEventLoop).run()
+        do {
+            try requestPromise.futureResult.wait()
+            XCTFail("Did not throw")
+        } catch is NIOHTTP2Errors.StreamClosed {
+            // ok
+        } catch {
+            XCTFail("Unexpected error: \(error)")
+        }
```

Comment:
Let's replace this with:

```swift
XCTAssertThrowsError(try requestPromise.futureResult.wait()) { error in
    XCTAssertTrue(error is NIOHTTP2Errors.StreamClosed)
}
```

Reply:
I'm going to replace it also in `testBasicPipelineCommunicates` since I copy and pasted the code for this test from there and applied some small modifications.